### PR TITLE
config: deprecate allow-auto-random option

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Fix https://github.com/pingcap/tiup/issues/598

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed: Add `allow-auto-random` to deprecate config option to avoid config-check failed.

### Check List 

Tests 

- Unit test
- Integration test
- Manual test

Now it will return `0` if we provide allow-auto-random configuration.

```console
$ ./bin/tidb-server -config ~/.config/tidb/config.toml.example -config-check true
config file /home/tangenta/.config/tidb/config.toml.example contained unknown configuration options: experimental.allow-auto-random
config check successful
```

Side effects

### Release note 

- No release note.

### CODE REVIEW

File: config/config.toml.example [hunk 1]
```diff
@@ -439,6 +439,8 @@ history-size = 24
 # experimental section controls the features that are still experimental: their semantics,
 # interfaces are subject to change, using these features in the production environment is not recommended.
 [experimental]
+# enable column attribute `auto_random` to be defined on the primary key column.
```
Comment (bb7133) [inline config/config.toml.example:442]:
```suggestion
# deprecated, it is not required to enable the `auto_random` feature.
```

File: config/config.go [hunk 2]
```diff
@@ -553,6 +553,8 @@ type IsolationRead struct {
 // Experimental controls the features that are still experimental: their semantics, interfaces are subject to change.
 // Using these features in the production environment is not recommended.
 type Experimental struct {
+	// Whether enable the syntax like `auto_random(3)` on the primary key column.
```
Comment (bb7133) [inline config/config.go:556]: // Deprecated, it is not required to enable the `auto_random` feature.

File: config/config.go [hunk 3]
```diff
@@ -553,6 +553,8 @@ type IsolationRead struct {
 // Experimental controls the features that are still experimental: their semantics, interfaces are subject to change.
 // Using these features in the production environment is not recommended.
 type Experimental struct {
+	// Deprecated, it is not required to enable the auto_random feature.
+	AllowAutoRandom bool `toml:"allow-auto-random" json:"allow-auto-random"`
```
Comment (jackysp) [inline config/config.go:557]: I think we don't need to add it back.

**Comment by tangenta** [comment]
/run-check_dev_2
```
[2020-07-16T04:09:59.409Z] ----------------------------------------------------------------------
[2020-07-16T04:09:59.409Z] FAIL: http_handler_test.go:1058: HTTPHandlerTestSuite.TestAllServerInfo
[2020-07-16T04:09:59.409Z] 
[2020-07-16T04:09:59.409Z] http_handler_test.go:1080:
[2020-07-16T04:09:59.409Z]     c.Assert(ok, Equals, true)
[2020-07-16T04:09:59.409Z] ... obtained bool = false
[2020-07-16T04:09:59.409Z] ... expected bool = true
```

**Pull request merged**